File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/apps/permissions/AppsPermissionsInfoRequest.java
package com.github.seratch.jslack.api.methods.request.apps.permissions;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

public class AppsPermissionsInfoRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `none`
     */
    private String token;

    AppsPermissionsInfoRequest(String token) {
        this.token = token;
    }

    public static AppsPermissionsInfoRequestBuilder builder() {
        return new AppsPermissionsInfoRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public static class AppsPermissionsInfoRequestBuilder {
        private String token;

        AppsPermissionsInfoRequestBuilder() {
        }

        public AppsPermissionsInfoRequest.AppsPermissionsInfoRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public AppsPermissionsInfoRequest build() {
            return new AppsPermissionsInfoRequest(token);
        }

        public String toString() {
            return "AppsPermissionsInfoRequest.AppsPermissionsInfoRequestBuilder(token=" + this.token + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/apps/permissions/AppsPermissionsInfoRequest.kt
package work.socialhub.kslack.api.methods.request.apps.permissions

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest
import kotlin.js.JsExport

@JsExport
class AppsPermissionsInfoRequest(
    /** Authentication token. Requires scope: `none` */
    override var token: String?
) : SlackApiRequest, FormRequest {

    override fun toMap(): Map<String, Any> {
        return mutableMapOf()
    }
}